API/NoDocs/Microsoft.Extensions.Hosting/AspireAzureEFCoreCosmosDBExtensions.cs

```diff
@@ -0,0 +1,14 @@
+// Assembly 'Aspire.Microsoft.EntityFrameworkCore.Cosmos'
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Aspire.Microsoft.EntityFrameworkCore.Cosmos;
+using Microsoft.EntityFrameworkCore;
+
+namespace Microsoft.Extensions.Hosting;
+
+public static class AspireAzureEFCoreCosmosDBExtensions
+{
+    public static void AddCosmosDbContext<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties)] TContext>(this IHostApplicationBuilder builder, string connectionName, string databaseName, Action<EntityFrameworkCoreCosmosDBSettings>? configureSettings = null, Action<DbContextOptionsBuilder>? configureDbContextOptions = null) where TContext : DbContext;
+    public static void EnrichCosmosDbContext<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties)] TContext>(this IHostApplicationBuilder builder, Action<EntityFrameworkCoreCosmosDBSettings>? configureSettings = null) where TContext : DbContext;
```

Comment:
We have somewhat analogous methods called ConfigureX, but they're also tied to the options pattern.  Probably not enough to cause confusion though?

Reply:
What does it do if you don't pass the action?  It reads from config.

Reply:
Seems like it's implicitly accepting the root `IConfiguration` as an argument.  If we made it explicit, it would be more obvious that it's reading from config and give the option of passing a different config.

Reply:
Alt: Augment.